### PR TITLE
swaylock-effects: init at 1.6.0

### PR DESCRIPTION
**Pull request by bram209**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
Don't forget to add yourself to the maintainers list in a separate commit titled `maintainers: add bram209` (this is why CI is failing). See [maintainers-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) for instructions and examples.

**Comment by bram209** [comment]
thanks @cole-h, added myself to the list : )

File: pkgs/applications/window-managers/sway/lock-effects.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "swaylock-effects";
+  version = "1.6.0";
+
+  src = fetchFromGitHub {
+    owner = "mortie";
+    repo = "swaylock-effects";
+    rev = "edac98443bffe351a5e095cd625c95c6e8d6bfe5";
```
Comment (cole-h) [inline pkgs/applications/window-managers/sway/lock-effects.nix]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/window-managers/sway/lock-effects.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  buildInputs = [ wayland wayland-protocols libxkbcommon cairo gdk-pixbuf pam ];
+
+  mesonFlags = [
+    "-Dpam=enabled" "-Dgdk-pixbuf=enabled" "-Dman-pages=enabled"
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Screen locker with effects for Wayland";
+    longDescription = ''
+      swaylock-effects is a screen locking utility with built-in screenshots and image manipulation effects for Wayland compositors.
```
Comment (cole-h) [inline pkgs/applications/window-managers/sway/lock-effects.nix]:
```suggestion
      swaylock-effects is a screen locking utility with built-in screenshots
      and image manipulation effects for Wayland compositors.
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -19711,6 +19711,7 @@ in
   swaylock = callPackage ../applications/window-managers/sway/lock.nix { };
 
   swaylock-fancy = callPackage ../applications/window-managers/sway/lock-fancy.nix { };
+  swaylock-effects = callPackage ../applications/window-managers/sway/lock-effects.nix { };
```
Comment (cole-h) [inline pkgs/top-level/all-packages.nix]:
```suggestion

  swaylock-effects = callPackage ../applications/window-managers/sway/lock-effects.nix { };
```

`swaylock-fancy` isn't right below `swaylock`, so no need (IMHO) for `swaylock-effects` to be right under `swaylock-fancy`

File: pkgs/applications/window-managers/sway/lock-effects.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    '';
+    inherit (src.meta) homepage;
```
Comment (cole-h) [inline pkgs/applications/window-managers/sway/lock-effects.nix]: I don't know if this is frowned upon in general or not, but I would still prefer to see:
```suggestion
    homepage = "https://github.com/mortie/swaylock-effects";
```

**Review by cole-h — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (cole-h) [reply pkgs/top-level/all-packages.nix]: And I'd also put `-effects` before `-fancy`, just to keep it sorted alphabetically.

**Comment by bram209** [comment]
@cole-h thanks for the review, applied all your suggestions

**Review by cole-h — APPROVED** [review]
LGTM. Doesn't work for me (`pam_authenticate failed: authentication information unavailable`), but neither does normal `swaylock`, and it's likely because I'm not on NixOS.

```
[3 built, 1 copied (0.0 MiB), 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/81306
7 package marked as broken and skipped:
crossLibcStdenv linuxPackages_4_4.sysdig linuxPackages_4_9.sysdig linuxPackages_hardkernel_4_14.sysdig synfigstudio telepathy-salut telepathy_salut

1 package built:
swaylock-effects
```

**Comment by bram209** [comment]
Try with
```
security.pam.services.swaylock = {};
```

**Comment by doronbehar** [comment]
There's an evaluation error by @GrahamcOfBorg : https://gist.github.com/GrahamcOfBorg/fca3a8a7a7559c3836035ab2e357fa17 .

**Comment by cole-h** [comment]
@GrahamcOfBorg eval

I think this was from when ShellCheck was marked as broken in a Haskell packages update.

File: pkgs/applications/window-managers/sway/lock-effects.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "swaylock-effects";
+  version = "1.6.0";
+
+  src = fetchFromGitHub {
+    owner = "mortie";
+    repo = "swaylock-effects";
+    rev = "v${version}";
```
Comment (cole-h) [inline pkgs/applications/window-managers/sway/lock-effects.nix]:
```suggestion
    rev = "v1.6-0";
```

Tag is actually `v1.6-0` for whatever reason.

**Review by cole-h — CHANGES_REQUESTED** [review]

File: pkgs/applications/window-managers/sway/lock-effects.nix (hunk 5 above)
Comment (doronbehar) [reply pkgs/applications/window-managers/sway/lock-effects.nix]: Then maybe `version` should be changed instead?

Comment (bram209) [reply pkgs/applications/window-managers/sway/lock-effects.nix]: Changed `version` to `v1.6-0`

File: pkgs/applications/window-managers/sway/lock-effects.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "swaylock-effects";
+  version = "1.6-0";
+
+  src = fetchFromGitHub {
+    owner = "mortie";
+    repo = "swaylock-effects";
+    rev = "v${version}";
+    sha256 = "11nap4xylq48pv7qaqxdcz3lbb2hyki8r996rzfzpw467mkxhsr5";
```
Comment (cole-h) [inline pkgs/applications/window-managers/sway/lock-effects.nix]:
```suggestion
    sha256 = "15lshqq3qj9m3yfac65hjcciaf9zdfh3ir7hgh0ach7gpi3rbk13";
```

Looks like the hash changed, causing build to fail.

**Comment by evils** [comment]
closing since #84178 merged nearly the same

**Pull request closed without merging**